.github/workflows/build-lint-test.yml

```diff
@@ -17,25 +17,25 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        node-version: [18.x, 20.x]
+        node-version: [18.x, 20.x, 22.x]
     steps:
-      - uses: actions/checkout@v4
-      - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v4
+      - name: Checkout and setup environment
+        uses: MetaMask/action-checkout-and-setup@v1
         with:
+          is-high-risk-environment: false
           node-version: ${{ matrix.node-version }}
-          cache: yarn
-      - run: yarn --immutable
-      - name: Cache "@metamask/snaps-execution-environments" build
+          cache-node-modules: ${{ matrix.node-version == '20.x' }}
+      - name: Prepare "@metamask/snaps-execution-environments" build
+        run: yarn workspace @metamask/snaps-execution-environments run build:lavamoat
+      - name: Save "@metamask/snaps-execution-environments" build
         id: cache-snaps-execution-environments-build
-        uses: actions/cache@v4
+        uses: actions/upload-artifact@v4
         with:
+          name: snaps-execution-environments-build-${{ runner.os }}-${{ matrix.node-version }}-${{ github.sha }}
+          retention-days: 1
           path: |
+            .nvmrc
```

Comment:
Why do we need to persist the NVM file?

Reply:
It's a workaround to get `action/upload-artifact` to preserve the entire file structure. We basically just need some file that doesn't change (within a workflow run) in the root folder, previously the "stub" file, but I figured we can just also use another random file. By default, it would only keep the part after a wildcard (`packages/*/dist` -> `*/dist`) or the last part (`packages/snaps-execution-environments/dist/browserify` -> `browserify`) of the file structure.